### ISSUE TITLE
Archive resource doesn't do anything if neither `creates` or `checksum` are set

### ISSUE THREAD

**Issue by joshuaspence** [voxpupuli/puppet-archive#229]
It seems that if neither `creates` nor `checksum` are set then the `archive` resource silently just does nothing:

```
  # `File[$archive_path]` is defined elsewhere.

  archive { $archive_path:
    extract       => true,
    extract_path  => $extract_path,
  }
```

**Comment by ffrank** [comment]
Yes, it's unfortunate. This is a side-effect of the implementation of the resource type. In order to fix this, we would have to go back to the chalk board and come up with a totally different structure of how the type operates.

I'm open to suggestions, but I feel that the current compromise is adequate, all things considered.
